evm-contracts/src/v0.6/dev/AggregatorProxy.sol

```diff
@@ -281,13 +353,45 @@ contract AggregatorProxy is AggregatorInterface, AggregatorV3Interface, Owned {
     setAggregator(_aggregator);
   }
 
+
   /*
    * Internal
    */
 
   function setAggregator(address _aggregator)
     internal
   {
-    aggregator = AggregatorV3Interface(_aggregator);
+    currentPhase.id++;
+    phaseAggregators[currentPhase.id] = _aggregator;
+    currentPhase.aggregator = _aggregator;
+  }
+
+  function addPhase(
+    uint256 _phase,
+    uint256 _originalId
+  )
+    internal
+    view
+    returns (uint256)
+  {
+    return (_originalId & REQUEST_ID_MASK) | _phase.mul(PHASE_BASE);
+  }
+
+  function getRoundIdPhaseIdAndAggregator(
+    uint256 _requestId
+  )
+    internal
+    view
+    returns (uint256, uint16, address)
+  {
+    uint16 requestPhaseId = uint16((PHASE_MASK & _requestId) >> PHASE_OFFSET);
+    uint256 requestRoundId = _requestId & REQUEST_ID_MASK;
+
+    return (
+      requestRoundId,
+      requestPhaseId,
+      phaseAggregators[requestPhaseId]
+    );
```

Comment:
I think that we can now return structs from internal functions without using `ABIEncoderV2`. Pros are that is cleaner to read and use, but it's potentially more costly. Consider using Phase struct here:

```suggestion
  function getRoundIdAndPhase(
    uint256 _requestId
  )
    internal
    view
    returns (uint256, Phase memory)
  {
    uint16 requestPhaseId = uint16((PHASE_MASK & _requestId) >> PHASE_OFFSET);
    Phase memory phase = Phase(requestPhaseId, phaseAggregators[requestPhaseId])
    uint256 requestRoundId = _requestId & REQUEST_ID_MASK;

    return (requestRoundId, phase);
```